(linting): formatted the code to remove ESlint errors:

osbornetunde opened the pull request:
Fixes #50  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` or the user who published the issue for help, in a comment below

@rexagod I'm currentingly having issues linting the transpiled "PublicLab.Grapher.js" in dist directory.

> We're happy to help you get this ready -- don't be afraid to ask for help!

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

osbornetunde committed:
formatted the code to remove errors:

IshaGupta18 commented:
This is a pretty big PR here! Amazing work! However, it would be best if @rexagod would approve this one. We should wait for his comment here. Thanks a ton!

osbornetunde commented:
That's cool. You are welcome

rexagod commented on .eslintrc.js:
```diff
@@ -13,5 +16,8 @@ module.exports = {
     },
     "rules": {
-    }
+      "no-console":0,
```
```suggestion
      "no-console": "off",
```
Just keeping things consistent!

osbornetunde replied:
for the `"no-console":` when I write `"off"` it throws an error that `0 = "off"`.  It means we will have to stick with `"no-console": 0`

rexagod commented on .eslintrc.js:
```diff
@@ -13,5 +16,8 @@ module.exports = {
     "rules": {
-    }
+      "no-console":0,
+      "no-unused-vars": "off",
```
We would actually like have this `on`.

osbornetunde replied:
If the `"no-console":` rule is set to` on`, it means I will have to remove all the `console.log()` statement in the file. Is that ok by you

osbornetunde replied:
Do you want both `"no-console":` and ` "no-unused-vars":`  **on** ?

rexagod replied:
Sorry, I meant `on` only for the `no-unused-vars`.

osbornetunde replied:
I think this is were the conflict is coming from. eslint has no `on` option, for `"no-unused-vars:"` to be it has to be set `"error"` which is the equivalent of `on`. Should I just remove `"no-unused-vars:"` from the eslint file to keep `on`.

rexagod commented on package.json:
```diff
@@ -27,16 +27,19 @@
   "dependencies": {
     "chai": "^4.2.0",
     "chartjs": "^0.3.24",
+    "file-saver": "^2.0.2",
```
Could you briefly elaborate on why and where has this been used?

osbornetunde replied:
In **view.js** file between lines **251 - 266**, there is a **saveAs** method called in the function below which wasn't defined, So I had to install **file-saver** and an import **saveAs** from it. File-saver is a JavaScript library for saving 
**saveAsImageFunction(xx){
        console.log("entered image");
        var x=new Date();
        var timestamp=x.getTime();
        var temp=xx;
        temp="#"+temp;
        console.log(temp,"omg");
        // var temp2=temp.slice(0,temp.length-5);
        // console.log(temp2);
        console.log(document.getElementById(xx));
        var tt=document.getElementById(xx);
        $(temp).get(0).toBlob(function(blob) {
            **saveAs**(blob, "chart"+timestamp);
        });
}**

rexagod replied:
Oh, that makes sense. Nice work!

rexagod commented on package.json:
```diff
@@ -27,16 +27,19 @@
     "mocha": "^6.1.4",
-    "papaparse": "^4.6.3"
+    "papaparse": "^4.6.3",
+    "xlsx": "^0.14.3"
```
Could you briefly elaborate on why and where has this been used?

osbornetunde replied:
I noticed the **handleFileSelectstring** function on line **52-73** in **View.js** file and the parse method on line **38-53** in **CsvParser.js** needed **papa** to parse a **csv file** but it wasn't installed, so I installed **papaparse** and imported **papa** for parsing the csv file.

**handleFileSelectstring**(val){
        console.log("i am at csv string handler");
        var csv_string = val.split("\n");
        var mat=[];
        for (var i=0;i<csv_string.length;i++){
            if (csv_string[i]=="" || csv_string[i]==" "){
            continue;
            }
            var dataHash=Papa.parse(csv_string[i],{
            dynamicTyping: true,
            comments: true
            });
            mat[i]=dataHash['data'][0];
        }
        this.csvFile=mat;
        let self = this;
        document.getElementById(this.uploadButtonId).onclick = (e) => {
            console.log("i am uploading");
            self.csvParser = new CsvParser(self.csvFile, self.elementId,"csvstring");
        };

    }

**parse**(functionParameter){
        var count = 0;
        Papa.parse(this.csvFile, {
            download: true,
            dynamicTyping: true,
            comments: true,
            step: (row) => {
                this.csvMatrix[count] = row.data[0];
                count += 1;
            },
            complete: () => {
                //calling a function to determine headers for columns
                this.startFileProcessing(functionParameter);
            }
        });
    }

**"XLSX"** was needed to create an excel sheet in the **createSheet** method in line **297-319** of **View.js**, so I had to install **"xlsx" : "^0.14.3**, then import XLSX from it

**createSheet**(){
        var wb = XLSX.utils.book_new();
        wb.Props = {
                Title: "New Spreadsheet"+this.elementId,
                CreatedDate: new Date()
        };
        
        wb.SheetNames.push("Sheet"+this.elementId);
        var ws_data = this.csvParser.completeCsvMatrixTranspose;
        var ws = XLSX.utils.aoa_to_sheet(ws_data);
        wb.Sheets["Sheet"+this.elementId] = ws;
        var wbout = XLSX.write(wb, {bookType:'xlsx',  type: 'binary'});
        function s2ab(s) {
    
                var buf = new ArrayBuffer(s.length);
                var view = new Uint8Array(buf);
                for (var i=0; i<s.length; i++) view[i] = s.charCodeAt(i) & 0xFF;
                return buf;
                
        }
        saveAs(new Blob([s2ab(wbout)],{type:"application/octet-stream"}), 'newSpreadsheet'+this.elementId+'.xlsx');

    }

rexagod replied:
Hmm. You've verified this local, right? Did everything work without any issues? Sorry, but I need to make sure we're good here since this seems to be a large refactor.

osbornetunde replied:
How do I verify it on my local? I can find how to run the file on my localhost.

rexagod replied:
I suppose @IshaGupta18 can help you set up the dev environment on your local.

osbornetunde replied:
Ok. I will very much appreciate that

rexagod replied:
Pinging @IshaGupta18 for a look here, thanks!

IshaGupta18 replied:
When you clone the repo, do npm install to install the
dependencies, and do npm run build, to see the changes in the main file. This is all I think for the dependencies. @namangupta01 will be able to guide you better on this one. Thanks a lot!

IshaGupta18 replied:
Oh I very much wrote this msg a few hours ago, I don't know how it wasn't posted. Terribly sorry!

osbornetunde replied:
I ran npm run build, I only see changes in files placed in the dist directory

rexagod replied:
Yeah, its the bundled up distributable that gets generated by all the `src` files and into one huge file in the dist folder. Are the changes as expected when you are running the app?

osbornetunde replied:
Do you mean when I run the app using **npm run build**? If so, yes, the changes are the ones expected.

rexagod commented on spec/js/sample_test.js:
```diff
@@ -1,9 +1,9 @@
-    it('Should return Mocha Testing', function () {
-        result = sampleTest();
+assert.describe("Sample Test", function () {
+    assert.it('Should return Mocha Testing', function () {
```
Can we do away with calling these functions from `assert`? I think it'd be better if we reverted this back to avoid calling `assert.fn` instead of just `fn`, just so it stays a bit more cleaner.

osbornetunde replied:
I have reverted it back, I need your go ahead on the necessary changes I need to make in the package.json file before pushing it. Thanks

rexagod replied:
Please do!

rexagod commented on src/CsvParser.js:
```diff
@@ -74,35 +75,35 @@ class CsvParser{
     }
 
-    //makes a 2D matrix with the transpose of the CSV file, each column having the same index as its column heading
+    //makes a 2D matrxx with the transpose of the CSV file, each column having the same index as its column heading
```
```suggestion
    //makes a 2D matrix with the transpose of the CSV file, each column having the same index as its column heading
```

rexagod reviewed (changes requested):
This is some really nice work, @osbornetunde. Just a few changes, which I mentioned above, after which we can surely get this merged. Thanks!

osbornetunde committed:
Update .eslintrc.js

Co-Authored-By: Pranshu Srivastava <rexagod@gmail.com>

osbornetunde committed:
Update src/CsvParser.js

Co-Authored-By: Pranshu Srivastava <rexagod@gmail.com>

osbornetunde committed:
(linting): implemented the corrections
-worked on the package.json
-removed the all unused"e"
-reverted sample.js back to its original state

osbornetunde committed:
resolved merge conflicts

namangupta01 commented:
Hi @osbornetunde, wow big pr. Nice work. There are conflicts in some files can you please fix them. And what do you think @jywarren , @rexagod and @IshaGupta18 should be not included dist/ in pr and only admin build and push to dist/ what do you guys think?

osbornetunde committed:
lock file

osbornetunde commented:
Yes, I couldn't locate it on my local after doing a git pull. But it seems to be from the eslintrc.js, the rule `"no-unused-vars:"` has to be `"error"` which is the equivalent of `on`. I have updated and pushed it, so I need @rexagod  to look into it because it was one of the changes He requested for. Thanks

osbornetunde commented:
Yes, I couldn't locate it on my local after doing a git pull. But it seems to be from the eslintrc.js, the rule `"no-unused-vars:"` has to be `"error"` which is the equivalent of `on`. I have updated and pushed it, so I need @rexagod  to look into it because it was one of the changes He requested for. Thanks

osbornetunde committed:
resolved conflicts

osbornetunde committed:
updated output of "npm run build" in dist

osbornetunde commented:
@namangupta01 I have resolved the conflicts

rexagod commented:
@namangupta01 I'm okay with the contributors (non-admins) pushing `dist` files into their PR branches.

Also, @IshaGupta18 @namangupta01 Let's get this one merged, so please let @osbornetunde know about the updation in the changes, if any, so we can accelerate things here if the code looks good to you. Thanks!

IshaGupta18 commented:
This does look good to me! Great work here. I think from here on, we can
ignore the dist files!

On Fri, Jun 28, 2019, 9:38 PM Pranshu Srivastava <notifications@github.com>
wrote:

> @namangupta01 <https://github.com/namangupta01> I'm okay with the
> contributors (non-admins) pushing dist files into their PR branches.
>
> Also, @IshaGupta18 <https://github.com/IshaGupta18> @namangupta01
> <https://github.com/namangupta01> Let's get this one merged, so please
> let @osbornetunde <https://github.com/osbornetunde> know about the
> updation in the changes, if any, so we can accelerate things here if the
> code looks good to you. Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/simple-data-grapher/pull/56?email_source=notifications&email_token=AJXHQZ5UGRXUTLP4ZVTL533P4YZO3A5CNFSM4H2QFYI2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY2P2XQ#issuecomment-506789214>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJXHQZZTS2XQGKMF7BOOWITP4YZO3ANCNFSM4H2QFYIQ>
> .
>

IgorWilbert commented:
Nice work @osbornetunde, thanks a lot!!!

osbornetunde commented:
@IgorWilbert you are welcome

osbornetunde committed:
resolved conflicts

osbornetunde committed:
removed no-used var from .eslint.js

osbornetunde committed:
updated my local branch

dsuarezlogans commented:
Hello is this PR going to be merge? Because I work in a fix that I would like to PR
